cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxDownloader.java

```diff
@@ -285,113 +83,186 @@ public void run() {
-                        task.handle = downloader._httpClient.head(Cocos2dxHelper.getActivity(), url, headers, null, task.handler);
-                        break;
-                    }
-
-                    // file task
-                    File tempFile = new File(path + downloader._tempFileNameSufix);
-                    if (tempFile.isDirectory()) break;
 
-                    File parent = tempFile.getParentFile();
-                    if (!parent.isDirectory() && !parent.mkdirs()) break;
-
-                    File finalFile = new File(path);
-                    if (tempFile.isDirectory()) break;
+                    final Request request = builder.build();
+                    task = downloader._httpClient.newCall(request);
+                    task.enqueue(new Callback() {
+                        @Override
+                        public void onFailure(Call call, IOException e) {
+                            downloader.onFinish(id, 0, e.toString(), null);
+                        }
 
-                    task.handler = new FileTaskHandler(downloader, id, tempFile, finalFile);
-                    long fileLen = tempFile.length();
-                    if (supportResuming && fileLen > 0) {
-                        // continue download
-                        listHeaders.add(new BasicHeader("Range", "bytes=" + fileLen + "-"));
-                        headers = listHeaders.toArray(new Header[listHeaders.size()]);
-                    }
-                    else if (fileLen > 0) {
-                        // Remove previous downloaded context
-                        try {
-                            PrintWriter writer = new PrintWriter(tempFile);
-                            writer.print("");
-                            writer.close();
+                        @Override
+                        public void onResponse(Call call, Response response) throws IOException {
+                            InputStream is = null;
+                            byte[] buf = new byte[4096];
+                            FileOutputStream fos = null;
+
+                            try {
+                                long total = response.body().contentLength();
+                                if (path.length() > 0 && !_resumingSupport.containsKey(host)) {
+                                    if (total > 0) {
+                                        Cocos2dxDownloader.setResumingSupport(host, true);
+                                    } else {
+                                        Cocos2dxDownloader.setResumingSupport(host, false);
+                                    }
+                                }
+
+                                long current = downloadStart;
+                                is = response.body().byteStream();
+
+                                if (path.length() > 0) {
+                                    fos = new FileOutputStream(tempFile);
```

Comment:
这里是不是需要设置append:true
```suggestion
                                    fos = new FileOutputStream(tempFile, true);
```

Reply:
你说的是对的